File: frontend/src/app/core/models/trainer.ts
export interface Trainer {
    id?: number;
    firstName: string;
    address: string;
    phoneNbr: string;
    lastName: string;
    trainerId: string;
    email: string;
    dob: Date;
    maritalStatus: string;
    gender: string;
    canEqual?: boolean;
    hashCode?: boolean;
    toString?: boolean;
}

export interface TrainerBuilder {
    id: number;
    firstName: string;
    lastName: string;
    email: string;
    phoneNbr: string;
    address: string;
    gender: string;
    dob: Date;
    maritalStatus: string;
    trainerId: string;
}

File: src/main/java/com/EduTrack/EDUTRACK_Portfolio_MS/model/Trainer.java
package com.EduTrack.EDUTRACK_Portfolio_MS.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Trainer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private String trainerId;
    private String firstName;
    private String lastName;
    private LocalDate dob;
    private String gender;
    private String maritalStatus;
    private String email;
    private String phoneNumber;
    private String address;
}
